core/url.c

```diff
@@ -506,3 +506,8 @@ oio_url_to_json (GString *out, struct oio_url_s *u)
 	}
 }
 
+gboolean
+oio_url_check(const struct oio_url_s *u)
+{
+    return !u->path || g_utf8_validate(u->path, -1, NULL);
```

Comment:
You could also validate the format of the version, e.g. with `oio_str_is_number()`.